Semantic Scholar.js

```diff
@@ -118,15 +118,19 @@ function parseDocument(doc, url) {
 			item.issue = volumeAndIssue[1];
 		}
 		
-		if (rawData.hasPdf) {
-			let paperLink = rawData.links.filter(function (link) {
-				return link.linkType === 's2';
-			})[0].url;
+		if (rawData.hasPdf && (rawData.primaryPaperLink.linkType === 's2'
+			|| rawData.primaryPaperLink.linkType == 'arxiv')) {
 			item.attachments.push({
-				url: paperLink,
+				url: rawData.primaryPaperLink.url,
 				title: "Full Text PDF",
 				mimeType: 'application/pdf'
 			});
+			if (rawData.primaryPaperLink.linkType == 'arxiv') {
+				let arxivId = rawData.primaryPaperLink.url.match(/\d{4}\.\d{5}/);
+				if (arxivId.length >= 1) {
+					item.extra = 'arXiv: ' + arxivId[0];
```

Comment:
There could already be something in the extra field from bibtex translator, which we don't want to override  then. Can you change this into something like
```js
if (item.extra) {
   item.extra += "\narXiv: " + arxivId[0];
}
else {
   item.extra = 'arXiv: ' + arxivId[0];
}
```
(untested code)

Reply:
Thanks, added